### PR TITLE
#10272: Choice of OGC service type for single layer in CSW catalog

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR gives the ability for the user to choose which service type, when the records has support for multi OGC service in the dc references, to be used when adding a layer to map via catalog of service type 'csw'

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Enhancement

## Issue

**What is the current behavior?**
- #10272 

**What is the new behavior?**
User can choose the service from which the layer to be added to Map.
>[!NOTE]
Currently support only WFS as the additional OGC service

image

https://github.com/geosolutions-it/MapStore2/assets/26929983/1d836b1d-fa18-4a81-9b27-d02bdc5cf51b

>[!NOTE]
**TEST DATA**
 https://gn3-rndt.geo-solutions.it/geonetwork/srv/eng/csw (Layer name - unesco)
 https://geobretagne.fr/geonetwork/srv/fre/csw (many records has multiple ogc service)

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/api/catalog/WFS.js [hunk 1]
```diff
@@ -123,7 +123,18 @@ export const getCatalogRecords = ({records} = {}) => {
     return null;
 };
 
+const getLayerData = (record, options) => {
```
Comment (MV88) [inline web/client/api/catalog/WFS.js:126]: jsdoc

File: web/client/components/catalog/RecordItem.jsx [hunk 2]
```diff
@@ -135,47 +137,77 @@ class RecordItem extends React.Component {
         return crs && !isSRSAllowed(crs);
     }
 
+    onAddToMap = (record, serviceType = record.serviceType) => {
+        if (this.isSRSNotAllowed(record)) {
+            return this.props.onError('catalog.srs_not_allowed');
+        }
+        this.setState({ loading: true });
+        return API[serviceType].getLayerFromRecord(record, {
```
Comment (MV88) [inline web/client/components/catalog/RecordItem.jsx:145]: why this has been moved to the component?

is this part of logic also needed in other catalogs around mapstore, outside the viewer?

**Review by MV88 — CHANGES_REQUESTED** [review]
LGTM, just a couple of comments with some jsdoc

File: web/client/api/catalog/CSW.js [hunk 3]
```diff
@@ -136,6 +141,28 @@ const recordToLayer = (record, options) => {
...
+    if (hasAdditionalService) {
+        return {
+            additionalOGCServices: {
+                ...ADDITIONAL_OGC_SERVICES
+                    .map(serviceType => {
+                        const ogcReferences = parsedReferences[serviceType] ?? {};
+                        const {url, params: {name} = {}} = ogcReferences;
+                        return {[serviceType]: {
+                            url, name, references, ogcReferences, fetchCapabilities: true,
+                            boundingBox: getBoundingBox(record)
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply web/client/api/catalog/CSW.js:161]: WFS refers to `boundingBox` prop to assign to `bbox` on layer formulation

File: web/client/components/catalog/RecordItem.jsx (hunk 2 above)
Comment (dsuren1) [reply web/client/components/catalog/RecordItem.jsx:145]: It was already a part of the component, I merely added a function block to make the component cleaner. Only in viewer (geostory, context, map) except dashboard catalog which is handled in a separate issue #10303

**Review by MV88 — APPROVED** [review]

**Comment by MV88** [comment]
@ElenaGallo please test in DEV

**Comment by ElenaGallo** [comment]
Test passed, @dsuren1 please backport to 2024.01.xx. Thanks

**Pull request merged**